test/suites/ai-model-serving/ai-model-serving-online.robot

```diff
@@ -0,0 +1,126 @@
+    Oc Apply    -n ${NAMESPACE} -f ${OVMS_KSERVE_MANIFEST}
+
+Deploy OpenVINO Resnet Model
+    [Documentation]    Deploys InferenceService object to create Deployment and Service to serve the model.
+    ...    Also creates a Route to export the model endpoint outside the MicroShift cluster.
+
+    Oc Apply    -n ${NAMESPACE} -f ./assets/ai-model-serving/ovms-resources.yaml
+    Wait Until Keyword Succeeds    10x    60s
+    ...    Run With Kubeconfig    oc rollout status -n\=${NAMESPACE} deployment openvino-resnet-predictor
+
+Check If Model Is Ready
+    [Documentation]    Asks model server is model is ready for inference.
+    ${cmd}=    Catenate
+    ...    curl
+    ...    --fail
+    ...    -i ${DOMAIN}/v2/models/${MODEL_NAME}/ready
+    ...    --connect-to "${DOMAIN}::${USHIFT_HOST}:"
+    Wait Until Keyword Succeeds    10x    10s
+    ...    Run Command    ${cmd}
+
+Query Model Metrics Endpoint
+    [Documentation]    Makes a query against the model server metrics endpoint.
+
+    ${cmd}=    Catenate
+    ...    curl
+    ...    --fail
+    ...    --request GET
+    ...    ${DOMAIN}/metrics
+    ...    --connect-to "${DOMAIN}::${USHIFT_HOST}:"
+    ${output}=    Run Command    ${cmd}
+    Should Contain    ${output}    ovms_requests_success Number of successful requests to a model or a DAG.
+
+Prepare Request Data
+    [Documentation]    Executes a script that prepares a request data.
+
+    Run Command    bash -x assets/ai-model-serving/ovms-query-preparation.sh ${OVMS_REQUEST}
+
+Remove Tmp Data
+    [Documentation]    Remove temp data for this test.
+
+    Run Command    rm ${OVMS_REQUEST} ${OVMS_KSERVE_MANIFEST}
+
+Query Model Infer Endpoint
+    [Documentation]    Makes a query against the model server.
+
+    # Inference-Header-Content-Length is the len of the JSON at the begining of the request.json
+    ${cmd}=    Catenate
+    ...    curl
+    ...    --silent
+    ...    --fail
+    ...    --request POST
+    ...    --data-binary "@${OVMS_REQUEST}"
+    ...    --header "Inference-Header-Content-Length: 63"
+    ...    ${DOMAIN}/v2/models/${MODEL_NAME}/infer
+    ...    --connect-to "${DOMAIN}::${USHIFT_HOST}:"
+    ${output}=    Run Command    ${cmd}
+    ${result}=    Json Parse    ${output}
+    ${data}=    Set Variable    ${result["outputs"][0]["data"]}
+    # Following expression can be referred to as 'argmax': index of the highest element.
+    ${argmax}=    Evaluate    ${data}.index(max(${data}))
+
+    # Request data includes bee.jpeg file so according to the OpenVino examples,
+    # we should expect argmax to be 309.
+    # See following for reference
+    # - https://github.com/openvinotoolkit/model_server/tree/releases/2025/0/client/python/kserve-api/samples#run-the-client-to-perform-inference-1
+    # - https://github.com/openvinotoolkit/model_server/blob/releases/2025/0/demos/image_classification/input_images.txt
+    Should Be Equal As Integers    ${argmax}    309
+
+Run Command
```

Comment:
Looks similar to existing `Command Should Work`
```suggestion
Local Command Should Work
```

Reply:
I'll address this in the next commit.